agenta-cli/agenta/sdk/decorators/routing.py

```diff
@@ -247,6 +251,26 @@ async def wrapper_deployed(*args, **kwargs) -> Any:
         app.post(route_deployed, response_model=BaseResponse)(wrapper_deployed)
         ### ---------------- #
 
+        ### --- Update Middleware --- #
+        try:
+            global _AUTH_MIDDLEWARE  # pylint: disable=global-statement
+
+            if not _AUTH_MIDDLEWARE:
+                app.add_middleware(
+                    AuthorizationMiddleware,
+                    host=ag.DEFAULT_AGENTA_SINGLETON_INSTANCE.host,
+                    resource_id=ag.DEFAULT_AGENTA_SINGLETON_INSTANCE.app_id,
+                    resource_type="application",
```

Comment:
won't the app id here be deprecated. Isn't the ressource id the project?

Reply:
The `app_id` is used for now because the `api keys` and `configs` PRs have not been merged and this builds on top of `observability checkpoint 2`.
This would have to be updated to some other resource later, probably the `base_id` or something.
I'm not sure about the `project_id` because it won't appear anywhere.